astroid/nodes/scoped_nodes.py

```diff
@@ -1570,6 +1570,8 @@ def type(
                 return "classmethod"
             if self.name == "__init_subclass__":
                 return "classmethod"
+            if self.name == "__class_getitem__":
+                return "classmethod"
```

Comment:
@Pierre-Sassoulas Just noticed that `__class_getitem__` was added in Python `3.7`. Adding a `sys.version_info` check here doesn't make much sense. It would block the usage of `py-version` for it in `pylint`. However we might want to consider updating the `getitem` logic (in pylint). The test case I wrote in https://github.com/PyCQA/pylint/pull/5354 shouldn't pass if `py-version = 3.6`.